storage/remote/client.go

```diff
@@ -147,6 +147,9 @@ func NewWriteClient(name string, conf *ClientConfig) (WriteClient, error) {
 
 type RecoverableError struct {
 	error
+	isRateLimiting bool
```

Comment:
I don't think `isRateLimiting` is necessary. A 0 value for `retryAfter` will mean to use the built in backoff instead.